### PR TITLE
Ability to Delete Project from CLI

### PR DESCRIPTION
**Pull request by ndneighbor**
Added `railway delete` 

If you provide a project uuid, it will delete the project directly- if you don't it will allow you to browse your account's projects to delete.

Help appreciated on improving the copy on error/success.

Thanks to @TaroballzChen for spearheading this improvement, all I did was debugging.

### CODE REVIEW

File: cmd/delete.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+		return err
+	}
+
+	if isLoggedIn {
+		return h.deleteFromAccount(ctx, req)
+	} else {
+		return h.deleteFromID(ctx, req)
+	}
+
+}
```
Comment (JakeCooper) [inline cmd/delete.go:45]: Delete should use a prompt which requires you to type in the name of the project

Comment (ndneighbor) [reply cmd/delete.go:45]: Good point- will be sure to add that to the flow.

Comment (TaroballzChen) [reply cmd/delete.go:45]: Thank for your attention

File: cmd/delete.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	isLoggedIn, err := h.ctrl.IsLoggedIn(ctx)
+	if err != nil {
+		return err
+	}
+
+	if isLoggedIn {
+		return h.deleteFromAccount(ctx, req)
+	} else {
+		return h.deleteFromID(ctx, req)
+	}
```
Comment (JakeCooper) [inline cmd/delete.go:43]: Don't need the else given this is the last function

```suggestion
	return h.deleteFromID(ctx, req)
```

File: cmd/delete.go [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+	"github.com/railwayapp/cli/uuid"
+)
+
+func (h *Handler) Delete(ctx context.Context, req *entity.CommandRequest) error {
+	user, err := h.ctrl.GetUser(ctx)
+	if err != nil {
+		return err
+	}
+	if user.Has2FA {
+		fmt.Printf("Your account has 2FA enabled, you must delete your project on the Dashboard.")
```
Comment (FarazPatankar) [inline cmd/delete.go:18]: Maybe add a link to `railway.app/dashboard` which they can just click?

File: cmd/delete.go [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+}
+
+func (h *Handler) deleteFromID(ctx context.Context, req *entity.CommandRequest) error {
+	projectID, err := ui.PromptText("Enter your project id")
+	if err != nil {
+		return err
+	}
+
+	project, err := h.ctrl.GetProject(ctx, projectID)
+	print("Looks good")
```
Comment (FarazPatankar) [inline cmd/delete.go:94]: Format/remove?

File: cmd/delete.go [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+}
+
+func (h *Handler) deleteFromAccount(ctx context.Context, req *entity.CommandRequest) error {
+	projects, err := h.ctrl.GetProjects(ctx)
+	if err != nil {
+		return err
+	}
+
+	if len(projects) == 0 {
+		fmt.Printf("No Projects could be deleted.")
```
Comment (FarazPatankar) [inline cmd/delete.go:62]: No projects found?

**Review by FarazPatankar — APPROVED** [review]
A few comments about the copy but looks good otherwise.

One case I am not sure if we've covered is, does `getProjects` only return the projects the user is an `admin` of? If not, what happens if they try to delete a project where they're only a member and what error do they see (if any)?

**Comment by ndneighbor** [comment]
@FarazPatankar that might be worth testing, will get to that tomorrow

**Comment by ndneighbor** [comment]
Update, the delete won't go through if you are not authorized to do so- working as intended :)

**Pull request merged**